corehq/apps/smsforms/models.py

```diff
@@ -103,16 +104,31 @@ def close(self):
         if not self.session_is_open:
             return
 
-        self.mark_completed(False)
-
-        if self.submit_partially_completed_forms:
-            submit_unfinished_form(self)
+        try:
+            if self.submit_partially_completed_forms:
+                submit_unfinished_form(self)
+        finally:
+            # this needs to be called after the submission, but regardless of whether it succeeded
+            # thus the try/finally
+            self.mark_completed(False)
 
     def mark_completed(self, completed):
         self.session_is_open = False
         self.completed = completed
         self.modified_time = self.end_time = utcnow()
 
+        metrics_counter('commcare.smsforms.session_ended', 1, tags={
```

Comment:
Do you think there could be value to recording the app and form ids?

Reply:
I try to avoid logging that kind of stuff in datadog. FWIW we already have a lot of that info in the messaging event logs which fairly meticulously record each of the steps and present them to the user as debugging information. The datadog metrics here are for me to be able to see broad trends site wide and/or for a given project.

Reply:
Makes sense, thanks for explaining.